### PR TITLE
fix: parser to ignore destination related query

### PR DESCRIPTION
**Pull request by deryrahman**

### CODE REVIEW

File: core/job/service/plugin_service.go [hunk 1]
```diff
@@ -121,9 +121,12 @@ func (p JobPluginService) GenerateDependencies(ctx context.Context, taskName job
 		return nil, err
 	}
 
-	// flatten
+	// flatten and ignore the destinatioon
```
Comment (irainia) [inline core/job/service/plugin_service.go:124]: we can remove this line

File: ext/parser/bq_parser.go [hunk 2]
```diff
@@ -20,7 +20,19 @@ var (
 			"|" +
 			"(?i)(?:WITH)\\s*(?:/\\*\\s*([a-zA-Z0-9@_-]*)\\s*\\*/)?\\s+`?([\\w-]+)\\.([\\w-]+)\\.([\\w-]+)`?\\s+(?:AS)" +
 			"|" +
-			"(?i)(?:VIEW)\\s*(?:/\\*\\s*([a-zA-Z0-9@_-]*)\\s*\\*/)?\\s+`?([\\w-]+)\\.([\\w-]+)\\.([\\w-]+)`?" +
+			"(?i)(?:VIEW)\\s*(?:/\\*\\s*([a-zA-Z0-9@_-]*)\\s*\\*/)?\\s+`?([\\w-]+)\\.([\\w-]+)\\.([\\w-]+)`?" + // can remove(?) already handled by CREATE clause below
```
Comment (irainia) [inline ext/parser/bq_parser.go:23]: we can remove the comment if not needed

**Review by irainia — CHANGES_REQUESTED** [review]

File: core/job/service/plugin_service.go [hunk 3]
```diff
@@ -121,9 +121,12 @@ func (p JobPluginService) GenerateDependencies(ctx context.Context, taskName job
...
 	resourceURNs := []job.ResourceURN{}
 	for _, r := range job.ResourceURNWithUpstreamsList(resources).Flatten() {
+		if r.URN == destinationURN {
+			continue
```
Comment (Mryashbhardwaj) [inline core/job/service/plugin_service.go:128]: how about we log such a scenarion, so that we can see , what all diffrent clauses are causing this

Comment (deryrahman) [reply core/job/service/plugin_service.go:128]: forgot to add it, yes we could log it as a warning

File: ext/parser/bq_parser.go [hunk 4]
```diff
@@ -49,10 +59,16 @@ func ParseTopLevelUpstreamsFromQuery(query string) []bigquery.ResourceURN {
 			ignoreUpstreamIdx, projectIdx, datasetIdx, nameIdx = 5, 6, 7, 8
 		case "with":
 			ignoreUpstreamIdx, projectIdx, datasetIdx, nameIdx = 9, 10, 11, 12
-		case "view":
```
Comment (Mryashbhardwaj) [inline ext/parser/bq_parser.go:52]: any reason why we are removing `view`

Comment (deryrahman) [reply ext/parser/bq_parser.go:52]: the case is already covered by this regex https://github.com/goto/optimus/pull/164/files#diff-ce0076ac87231740fec314d4c04407b96410d666a61fcd3df70cf75a24175f78R33

File: ext/parser/bq_parser_test.go [hunk 5]
```diff
@@ -99,7 +99,6 @@ func TestParseTopLevelUpstreamsFromQuery(t *testing.T) {
 				ExpectedResourceURNs: []bigquery.ResourceURN{
 					newBQResourceURN("project", "fire", "fly"),
 					newBQResourceURN("project", "maximum", "overdrive"),
-					newBQResourceURN("project", "maximum", "overdrive"), // the result should be unique(?))
```
Comment (Mryashbhardwaj) [inline ext/parser/bq_parser_test.go:102]: why we needed to remove this ?

Comment (deryrahman) [reply ext/parser/bq_parser_test.go:102]: it should not yield duplicate upstream, logic on transformer returns unique resource https://github.com/goto/transformers/blob/aeee206d8e2976d0e3ee55aacf87f419997a4009/task/bq2bq/upstream/parser.go#L84C24-L84C38

**Review by Mryashbhardwaj — APPROVED** [review]

**Review by irainia — APPROVED** [review]

**Pull request merged**